### ISSUE TITLE
Computed property is not reactive in tests when returning a nestedValue of an object prop

### ISSUE THREAD

**Issue by M3psipax** [vuejs/vue-test-utils#996]
### Version
1.0.0-beta.25

### Reproduction link
[https://codesandbox.io/s/j7jlr6q08y?previewwindow=tests](https://codesandbox.io/s/j7jlr6q08y?previewwindow=tests)

### Steps to reproduce
Just run the test. 

### What is expected?
The second test should pass. The computed property should reevaluate.

The vue component receives an object as prop. It has a computed property that returns a nested value from that object prop.
When changing this nested value, the computed property is not reevaluated in the test's wrapper vm. However, this does work in normal vue runtime.

### What is actually happening?
The computed property does not reevaluate.

**Comment by eddyerburgh** [comment]
There is an ongoing issue with our implementation of synchronous updates that will be resolved when Vue 2.5.18 is released.

For now there are two solutions, to set props you can use `setProps`:

```js
wrapper.setProps({
  obj: {
    nestedValue: "def"
  }
});
expect(wrapper.vm.cObjNestedValue).toEqual("def");
```

Or you can use a setTimeout to wait for Vue to perform an update:

```js
wrapper.vm.$props.obj = { nestedValue: "def" };
await new Promise(resolve => setTimeout(resolve));
expect(wrapper.vm.cObjNestedValue).toEqual("def");
```

**Comment by eddyerburgh** [comment]
Closing in favor of https://github.com/vuejs/vue-test-utils/issues/676
